src/lib/y2network/clients/auto.rb

```diff
@@ -116,6 +118,23 @@ def merge_current_config?
         !!Yast::Lan.autoinst.keep_install_network
       end
 
+      # Whatever is needed in /etc/hosts during AY installation
+      def update_etc_hosts
+        config = Yast::Lan.find_config(:yast)
+        return if !config
+
+        static_connections = config.connections.select(&:static?)
+        static_ips = static_connections.map { |c| c.ip.address.address.to_s }
```

Comment:
Why not assign the hostname to the connection? the writer should take care of updating the hosts entry, or is there some corner case? Isn't possible that otherwise the hostname is overwritten by the specific writer?

https://github.com/yast/yast-network/blob/SLE-15-SP2/src/lib/y2network/sysconfig/connection_config_writers/base.rb#L57

Reply:
i don't know ... I'm not that familiar with connections yet. But will give it a try.